### ISSUE TITLE
Discovering higher-level Patterns - Fundamentals

### ISSUE THREAD

**Issue by HyunkuKwon** [Computational-Content-Analysis-2020/Readings-Responses-Spring#26]
Post questions here for one or more of our fundamentals readings:

Manning, Christopher, Prabhakar Raghavan and Hinrich Schütze.  2008. “Flat Clustering” and “Hierarchical Clustering.” Chapters 16 and 17 from [Introduction to Information Retrieval](https://nlp.stanford.edu/IR-book/).

Blei, David. 2012. “[Probabilistic Topic Models](https://www.cs.princeton.edu/~blei/papers/Blei2012.pdf)”. Communications of the ACM 55(4):77-84.

**Comment by wanitchayap** [comment]
1. In my own experiences with topic models, sometimes some of the topics are too close together in semantic space. How can we guarantee that this won't happen? Does the algorithm as it already maximizes the differences between topics? Is it possible to eliminate the redundant topics and rerun the algorithm again with the rest of the topics fixed (that is the second run, k topics are no longer hidden variable)?

2. I totally agree with the relaxation of the BOW assumption. Besides the examples the authors give, do you think incorporating POS or other extractable linguistic information can improve LDA? Would using a pre-trained language model be helpful?

**Comment by WMhYang** [comment]
For **Manning, Christopher, Prabhakar Raghavan and Hinrich Schütze. 2008.**

In Chapter 17 the authors seem partial to hierarchical clustering when they compare it with flat clustering. The main reasons are that hierarchical clustering returns a structure of the clusters and it does not need a specific number of clusters. It seems that the drawback of hierarchical clustering, low efficiency, can be alleviated by algorithm design and large dataset so it is not a big deal. Does this mean that hierarchical clustering dominates flat clustering for most of times? Are there any cases where flat clustering could yield a better fit and a clearer intuition?

**Comment by nwrim** [comment]
For Blei (2012):

Every time I think about topic modeling, my concern is about interpretability. As shown in this review article, the process is usually: (1) we used n topics (2) for topic k, the top m words were: [a, b, c, ... ] (3) we believe this topic k corresponds to interpretable topic j. 

I think this has inherent problems of:
(1) sometimes the list of words for a topic does not provide a clear pattern and does not intuitively make sense. What are some approaches that can tune the topic model better if this happens?
(2) If we want to publish the findings, perhaps one of the reviewers do not like the label we but for a topic based on the list [a,b,c,...]. What could be some way to make them change their minds? What could be a supplementary material to make your label of topic stronger?

Another question:
Does doing statistical testing on the proportion of topics makes sense? For example, let's say we have a metric attached to authors or documents. Then, can we do statistical testing like topic_j_proportion_in_document ~ author_or_document_metric?
As a more concrete example, let's say we have the education level of the author (by school year). Then does it makes sense to do `topic_j_proportion_in_document ~ author_education_level`?

Final question:
Since LDA is inherently a statistical model, I think the result of the LDA could be (slightly) different everytime we run the algorithm (please correct me if I am wrong!), If this is the case, how do we report this in academic writing?

**Comment by timhannifan** [comment]
For Blei (2012):

I agree with @nwrim that interpretability is sometimes an issue with LDA. For me, this is largely a function of choosing a fixed number of topics at random. Is there a more methodological approach to finding the right n, or is this purely subjective?

For those considering LDA for their final project, I recommend looking at this package: [pyLDAvis](https://pyldavis.readthedocs.io/en/latest/readme.html). It produces very good visualizations for the outputs of these models in Python.

**Comment by Yilun0221** [comment]
LDA is a very interesting theme-catching tool. I used this in analyze the novel Pride and Prejudice and the results matched my understanding of the novel. My question here is, compared with tf-idf, what will be the advantages of LDA? Wha else information can we get? And also I think LDA will be limited to quite narrative text or text with proper grammers. It can be hard to use in some poems, or prose.

**Comment by timqzhang** [comment]
**For "Introduction to Information Retrieval"** Chapter 16

It is mentioned in Chapter 16 that the "Finding good seeds is even more critical for EM than for K-means", and the EM approach falls back to the classic question that the result is much based on the initial guess. It is not only the problem of EM, but also a lot of algorithms in the machine learning. I wonder is there any more state-of-the-art methods to alleviate the heavy dependence on the initial guess?

**Comment by linghui-wu** [comment]
As for the question mentioned by @Yilun0221, from my understanding, LDA not only takes into consideration of the frequency of a certain word in a topic but also accounts for the distribution of a particular topic given a document, the latter of which might be the extra layer of information we cannot obtain via tf-idf matrix.

My question is about how LDA handles the potentially different meanings of words? For instance,  some words might be borrowed by one topic from another. To what extent the algorithm may capture such a problem in practice?

**Comment by harryx113** [comment]
I noticed that there are nuanced differences in applications of LDA and word vector embeddings in that the LDA is more suitable for statistical relationship of occurrences and word embeddings extract more semantic information. Is this interpretation accurate?

**Comment by tianyueniu** [comment]
This might be a really basic question, but I don't entirely understand how exactly does LDA determine the assigned number of topics across documents in the first place?  Also, I assume that choosing different k (number of topics) can significantly affect our results, in that case, how should we choose k?

**Comment by Lesopil** [comment]
As social scientists, we are mostly interested in people and their interaction. If we are able to identify topics and themes using topic modeling that would never be able to be identified by people, is that still meaningful?

**Comment by jsgenan** [comment]
Blei, David. 2012. “Probabilistic Topic Models”
For Figure 5 in the paper, does it mean we can conclude two topic distributions with the dynamic topic model? If yes, how do we interpret the difference between those two sets of topics?

**Comment by liu431** [comment]
In my own project, I picked up the optimal topic number by both metrics and intuitions. However, this process cannot be automated when I apply my code to new data. Is there a better way to execute this?

**Comment by lyl010** [comment]
I have a question on Flat clustering(chapter 16 of introduction to information retrieval), the evaluation criterion purity needs two kinds of data: class of node and cluster of node, and I am wondering does the class of node suggests the true label of it? If so, I am confused because in clustering method we do not have the information of true label.
